### PR TITLE
Create a container element for search in standalone gnav

### PR DESCRIPTION
**Pull request by sonawanesnehal3**
Create a container element for search in standalone gnav

Resolves: [MWPW-157417](https://jira.corp.adobe.com/browse/MWPW-157417)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/?martech=off
- After: https://mwpw157417--milo--adobecom.hlx.page/?martech=off

**QA**
- https://adobecom.github.io/nav-consumer/navigation.html?searchEnabled=on&navbranch=mwpw157417&env=stage

### CODE REVIEW

**Review by bandana147 — APPROVED** [review]

File: libs/blocks/global-navigation/global-navigation.js [hunk 1]
```diff
@@ -813,6 +813,11 @@ class Gnav {
     analyticsValue: 'Logo',
   });
 
+  decorateClientSearch = () => {
+    this.elements.customSearch = toFragment`<div class="feds-client-search"></div>`;
+    return this.elements.customSearch;
+  };
+
```
[the remark this answers is not in the document]
Comment (mokimo) [reply libs/blocks/global-navigation/global-navigation.js:820]: To provide some options, either integrate a component that someone exports (similar to the unav) or simply write the new search component yourself.

[the remark this answers is not in the document]
Comment (salonijain3) [reply libs/blocks/global-navigation/global-navigation.js:820]: @mokimo Currently, we dont have the search component on adobe.com and this component is owned by adobe home, so thats why we are giving the search container. 

cc: @narcis-radu

**Review by mokimo — APPROVED** [review]
Had a small chat with @narcis-radu regarding this, seems like is the least evil option. Looks good in that case!

**Review by salonijain3 — APPROVED** [review]

File: libs/blocks/global-navigation/global-navigation.js [hunk 2]
```diff
@@ -813,6 +813,11 @@ class Gnav {
     analyticsValue: 'Logo',
   });
 
+  decorateClientSearch = () => {
+    this.elements.customSearch = toFragment`<div class="feds-client-search"></div>`;
+    return this.elements.customSearch;
```
Comment (overmyheadandbody) [inline libs/blocks/global-navigation/global-navigation.js:818]: It doesn't look like you're using `this.elements.customSearch` anywhere, so why not just
```suggestion
    return toFragment`<div class="feds-client-search"></div>`;
```

Alternatively, you could have the `searchEnabled` check here and return the element or `''` and then just call the method directly in `decorateMainNav`.

File: libs/navigation/navigation.js
[the remark this answers is not in the document]
Comment (overmyheadandbody) [reply libs/navigation/navigation.js:1]: I don't remember the context for this one, but having console errors might not be desirable. Would LANA logs not be enough?

[the remark this answers is not in the document]
Comment (bandana147) [reply libs/navigation/navigation.js:1]: This is for standalone Gnav in which we need to notify the consuming clients of the error.

**Comment by spadmasa** [comment]
Validations has ben done on the pr on the below url

https://adobecom.github.io/nav-consumer/navigation.html?searchEnabled=on&navbranch=mwpw157417&env=stage
image
results are updated in https://jira.corp.adobe.com/browse/MWPW-157417

File: libs/blocks/global-navigation/global-navigation.js [hunk 3]
```diff
@@ -822,6 +822,7 @@ class Gnav {
         ${isDesktop.matches ? '' : this.decorateSearch()}
         ${this.elements.mainNav}
         ${isDesktop.matches ? this.decorateSearch() : ''}
+        ${getConfig().searchEnabled === 'on' ? toFragment`<div class="feds-client-search"></div>` : ''}
```
Comment (overmyheadandbody) [inline libs/blocks/global-navigation/global-navigation.js:825]: One thing is not very clear to me. This feature is supposed to be relevant for the standalone Gnav, yet the config is added through `scripts.js` at a consumer level. Couldn't a config be passed directly to the standalone version? I feel like I'm missing something, but also want to make sure we're not polluting `scripts.js` with irrelevant properties.

Comment (bandana147) [reply libs/blocks/global-navigation/global-navigation.js:825]: @overmyheadandbody There is no separate config for standalone Gnav. The standalone version directly calls the init of global-navigation /global-footer and we set the miloLibs from `navigation.js` not through script.js.

**Pull request merged**